### ISSUE TITLE
Parent class methods not wrapped with exception code

### ISSUE THREAD

**Issue by tux-mind** [swig/swig#1599]
Hi, first of all, thanks for this awesome tool :)

I encountered a problem that I don't know how to workaround.
Methods from parent classes are not wrapped in the code I specify in the `%exception` directive.

I've attached a minimum working example of the problem.
I'm using swig 4.0.0.

The bug can be seen in the resulting wrapping cxx code: 
  - `_wrap_A_getRandomNumber_example_*` have the try/catch
  - `_wrap_B_getRandomNumber_example_*`  doesn't 

To obtain it just run `make` after unzipping [exceptions.zip](https://github.com/swig/swig/files/3430350/exceptions.zip).

All the best, tux-mind.

**Comment by wsfulton** [comment]
This is a problem peculiar to the Go module as this module creates extra wrappers for methods seen in base classes, whereas as the other SWIG modules create one wrapper and relies on the target language polymorphism to make the the function available in derived classes. @ianlancetaylor, I presume the usual approach was not possible in the Go implementation?

This is quite a big bug because `%exception` is a SWIG feature - `feature:except`.  It looks like any other SWIG features attached to Go functions will be ignored in the extra functions added into the Go derived classes. I can also see this with 'feature:new' a.k.a. `%newobject`.

**Comment by ianlancetaylor** [comment]
Yes, Go doesn't have class inheritance and doesn't have function overloading.  Every C++ type is represented as both an interface type and a C++ pointer type (represented in Go as a `uintptr`), with appropriate Go methods on the C++ pointer type.  So we have to write a wrapper for each method in Go, and they each have to call a wrapper in C++.  Maybe there is some more clever way to do this, but it's not obvious what it would be.

I guess that when we generate the extra C++ wrapper methods we need to somehow copy over the features.  Or something like that.  I think might be in `emitGoAction` in the case where `base` is not `nullptr`.  Maybe.
